AutoCollection/StatsBeat.ts

```diff
@@ -0,0 +1,218 @@
+    private _runtimeVersion: string;
+    private _os: string;
+    private _language: string;
+    private _ciIkey: string;
+    private _attach: string = Constants.StatsBeatAttach.sdk; // Default is SDK
+    private _features: number = Constants.StatsBeatFeature.NONE;
+    private _instrumentations: number = Constants.StatsBeatInstrumentation.NONE;
+
+    constructor(config?: Config) {
+        this._isInitialized = false;
+        this._config = config;
+        this._client = new TelemetryClient(STATS_BEAT_IKEY);
+    }
+
+    public enable(isEnabled: boolean) {
+        this._isEnabled = isEnabled;
+        if (this._isEnabled && !this._isInitialized) {
+            this._getCustomProperties();
+            this._isInitialized = true;
+        }
+        if (isEnabled) {
+            if (!this._handle) {
+                this._handle = setInterval(() => {
+                    this.trackStatsBeatMetrics()
+                }, this._collectionInterval);
+                this._handle.unref(); // Allow the app to terminate even while this loop is going on
+            }
+        } else {
+            if (this._handle) {
+                clearInterval(this._handle);
+                this._handle = null;
+            }
+        }
+    }
+
+    public isInitialized() {
+        return this._isInitialized;
+    }
+
+    public isEnabled() {
+        return this._isEnabled;
+    }
+
+    public setCodelessAttach() {
+        this._attach = Constants.StatsBeatAttach.codeless;
+    }
+
+    public addFeature(feature: Constants.StatsBeatFeature) {
+        this._features |= feature;
+    }
+
+    public removeFeature(feature: Constants.StatsBeatFeature) {
+        this._features &= ~feature;
+    }
+
+    public addInstrumentation(instrumentation: Constants.StatsBeatInstrumentation) {
+        this._instrumentations |= instrumentation;
+    }
+
+    public removeInstrumentation(instrumentation: Constants.StatsBeatInstrumentation) {
+        this._instrumentations &= ~instrumentation;
+    }
+
+    public countRequest(duration: number, success: boolean) {
+        if (!this.isEnabled()) {
+            return;
+        }
+        this._totalRequestCount++;
+        this._intervalRequestExecutionTime += duration;
+        if (success === false) {
+            this._totalFailedRequestCount++;
+        }
+        else {
+            this._totalSuccesfulRequestCount++;
+        }
+    }
+
+    public countException() {
+        if (!this.isEnabled()) {
+            return;
+        }
+        this._exceptionCount++;
+    }
+
+    public countThrottle() {
+        if (!this.isEnabled()) {
+            return;
+        }
+        this._throttleCount++;
+    }
+
+    public countRetry() {
+        if (!this.isEnabled()) {
+            return;
+        }
+        this._retryCount++;
+    }
+
+    public trackStatsBeatMetrics() {
+        this._trackRequestDuration();
+        this._trackRequestsCount();
+    }
+
+    private _trackRequestDuration() {
```

Comment:
traceRequestDurationAvg? it for average of the request duration in each 15 mins period.